### DISCUSSION TITLE
Django debug toolbar?

### DISCUSSION THREAD

**Discussion by ddahan** [vitalik/django-ninja#268] [Q&A] [answered]
Is there a way to make Django debug toolbar work with django ninja?
I noticed there is a [package dedicated to this for fastAPI](https://fastapi-debug-toolbar.domake.io/).
If there is a simple way to make it work, I think it should be documented, as it's very useful.
Thanks.

**Comment by vitalik** [comment]
It should work out of the box ? did you try  - what was the issue ?

**Reply by ddahan** [reply]
Yes. I installed Django Debug Toolbar (DDT), it works as expected (for example with the Django admin).
Then, I went to `http://localhost:8000/api/docs`. Loading this page triggers DDT.
Then, I tried to execute queries from this doc, using the "try it out" button. Unfortunately, despite the query is executed, this has no effect on DDT.

I guess the api docs use some sort of AJAX without changing the page URL, so DDT is unable to intercept the query.

![image](https://user-images.githubusercontent.com/1933516/140762799-63e5063d-dea7-4430-a804-252f7b0d96d8.png)

**Comment by vitalik** [comment, marked as answer]
@ddahan 

yeah, Debug Toolbar does not support ajax requests by default

I think you need to enable History Panel 
https://github.com/jazzband/django-debug-toolbar/pull/1250

https://django-debug-toolbar.readthedocs.io/en/latest/panels.html#debug_toolbar.panels.history.HistoryPanel

>This panel shows the history of requests made and allows switching to a past snapshot of the toolbar to view that request’s stats.

**Reply by ddahan** [reply]
Great, thanks. I'm indeed able to take a look to AJAX requests browsing this history.
